src/interactive-window/interactiveWindow.ts

```diff
@@ -782,6 +781,29 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
 
         return { cell, wasScrolled: shouldScroll };
     }
+    private async generateCodeAndAddMetadata(cell: NotebookCell, isDebug: boolean, kernel: IKernel) {
+        const metadata = getInteractiveCellMetadata(cell);
+        if (!metadata) {
+            return;
+        }
+        const useJupyterDebugger =
+            !isLocalConnection(kernel.kernelConnectionMetadata) ||
+            this.configuration.getSettings(undefined).useJupyterDebugger;
+
+        const generatedCode = this.codeGeneratorFactory
+            .getOrCreate(this.notebookDocument)
+            .generateCode(metadata, isDebug, useJupyterDebugger);
+
+        const newMetadata: typeof metadata = {
+            ...metadata,
+            generatedCode
+        };
+
+        const edit = new WorkspaceEdit();
+        const cellEdit = NotebookEdit.updateCellMetadata(cell.index, newMetadata);
```

Comment:
The IW code is generated only after we resolve the kernel.
Once we have a single debugger implementation we should be able to move this back into the place where the cell is created.